mlton/control/control-flags.sig

```diff
@@ -395,6 +395,15 @@ signature CONTROL_FLAGS =
        | Always
       val splitTypesBool: splitTypesBool ref
 
+      datatype staticAllocInternalPtrs =
+         All
+       | Static
+       | None
+      val staticAllocInternalPtrs: staticAllocInternalPtrs ref
+      val staticAllocArrays: bool ref
+      val staticAllocObjects: bool ref
+      val staticAllocVectors: bool ref
```

Comment:
In general, we should think of a grid of 4 possibilities:
Static/dynamic initialization (i.e. direct data / code) and static/dynamic allocation (i.e. text or heap). If staticAllocVectors is turned off, then WordXVectors will only be stored in the heap. Regardless, they will be initialized via fully static data. With the current setup, toRssa will turn constant WordXVectors into Statics, with allocation location depending on this setting, if it is on, they will be statically allocated, if it is off, they will be initialized to the heap.

staticAllocArrays is there because arrays in practice go through a substantially different code path. They use the separate .bss init, and are generally mutable. Realistically, a user will never want to deactivate these options unless they're encountering an issue, or they're doing something to the backend. WordXVectors are just generally safer.

Reply:
Also see https://github.com/MLton/mlton/pull/328#discussion_r323517751.

Reply:
Resolved by 4a49d4d and deb6081.